db.py

```diff
@@ -33,23 +39,26 @@ def get_people_for_restaurant(self, restaurant):
             names = cur.fetchall()
         return list(chain(*names))
 
+    # Throws DB.DBException if either the person name or one of the restaurant names is not found
+    # Thows sqlite3.IntegrityError if a mapping between person and one of the restaurants already exists
     def add_person_to_restaurants(self, person, restaurants_list):
-        #TODO Eran what happens if person exists?
         with self.con:
-            self.con.execute("INSERT INTO person_to_restaurant(person_id, restaurant_id) "
-                             "SELECT person.id, restaurant.id FROM person, restaurant "
-                             "WHERE restaurant.name in (" + ("?,"*len(restaurants_list))[:-1] + ") AND person.name=?;",
-                             list(chain(restaurants_list, [person])))
+            question_marks = ", ".join("?" * len(restaurants_list))
+            cur = self.con.execute("INSERT INTO person_to_restaurant(person_id, restaurant_id) "
+                                    "SELECT person.id, restaurant.id FROM person, restaurant "
+                                    "WHERE restaurant.name in (" + question_marks + ") AND person.name=?;",
+                                    list(chain(restaurants_list, [person])))
+            if cur.rowcount != len(restaurants_list):
+                raise DB.DBException("Person or one or more restaurants don't exist")
```

Comment:
Exception can be more informative by specifying the input (person %s or at least one of these restaurants %s ...)